TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/BlockLibrary/VirtualTexturePropertyBlockBuilder.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections.Generic;
+using UnityEditor.ShaderFoundry;
+using static UnityEditor.ShaderFoundry.VirtualTextureLayerAttribute;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    internal class VirtualTexturePropertyBlockBuilder : BasePropertyBlockBuilder
+    {
+        public class LayerData
+        {
+            internal string UniformName;
+            internal string DisplayName;
+            internal string TextureName;
+            internal LayerTextureType? TextureType = null;
+        }
+
+        const string VirtualTexturePropertyTypeName = "VTPropertyWithTextureType";
+        public int LayerCount = 2;
+        public int LayerToSample = 0;
+        
+        public List<LayerData> Layers = new List<LayerData>
+        {
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+        };
+
+        public static string GetName(int layerIndex, string layerValue, string propertyValue, string fieldName)
+        {
+            // Uniform and Display names follow a fallback scheme. It will use in order:
+            // - The layer specific value
+            // - The property attribute value
+            // - A default generated from the field name
+            var name = layerValue;
+            if (string.IsNullOrEmpty(name))
+            {
+                var backupName = propertyValue;
+                if (string.IsNullOrEmpty(backupName))
+                    backupName = fieldName;
+                name = $"{backupName}_Layer{layerIndex}";
+            }
+            return name;
+        }
+
+        public string GetUniformName(int layerIndex)
+        {
+            return GetName(layerIndex, Layers[layerIndex]?.UniformName, PropertyAttribute?.UniformName, FieldName);
+        }
+
+        public string GetDisplayName(int layerIndex)
+        {
+            return GetName(layerIndex, Layers[layerIndex]?.DisplayName, PropertyAttribute?.DisplayName, FieldName);
+        }
+
+        public string GetTextureDefault(int layerIndex)
+        {
+            return Layers[layerIndex]?.TextureName ?? PropertyAttribute?.DefaultValue ?? "\"\" {}";
+        }
+
+        public VirtualTexturePropertyBlockBuilder()
+        {
+            BlockName = "VirtualTextureProperty";
+            FieldName = "FieldVirtualTexture";
+            //PropertyAttribute = new PropertyAttributeData() { DefaultValue = "\"\" {}" };
```

Comment:
Is this expected to be uncommented soon, or is it documenting something?  If neither, it could be removed.

Reply:
No, I was commenting that out to make sure it was safe to delete and forgot to remove it